File: Autolock/Storage/CredentialsStore.swift
import Foundation

/// Loads and saves the single Bluelink account's credentials.
/// One phone, one Bluelink account -- no account switching, so a fixed key.
enum CredentialsStore {
    private static let account = "bluelink-primary"

    static func save(_ credentials: BluelinkCredentials) throws {
        try KeychainStore.save(credentials, account: account)
    }

    static func load() throws -> BluelinkCredentials? {
        try KeychainStore.load(BluelinkCredentials.self, account: account)
    }

    static func clear() throws {
        try KeychainStore.delete(account: account)
    }
}

File: android/app/src/main/java/com/sendprobe/autolock/storage/CredentialsStore.kt
package com.sendprobe.autolock.storage

import android.content.Context
import com.sendprobe.autolock.model.BluelinkCredentials

/**
 * Loads and saves the single Bluelink account's credentials. One phone,
 * one Bluelink account -- no account switching, so a fixed key.
 */
object CredentialsStore {
    private const val ACCOUNT = "bluelink-primary"

    fun save(context: Context, credentials: BluelinkCredentials) {
        SecureStore.save(context, credentials, ACCOUNT)
    }

    fun load(context: Context): BluelinkCredentials? =
        SecureStore.load(context, ACCOUNT)

    fun clear(context: Context) {
        SecureStore.delete(context, ACCOUNT)
    }
}
